### PR TITLE
DMT-1239-airflow_dev_eksRole_role_211908c

### PR DESCRIPTION
**Pull request by murad-ali-MoJ**

### CODE REVIEW

File: terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf [hunk 1]
```diff
@@ -334,3 +334,22 @@ data "aws_iam_policy_document" "airflow_prod_execution_role_policy" {
...
+########################Airflow dev EKS Policy###########################
+
+data "aws_iam_policy_document" "airflow_dev_eksRole_role_211908c_assume_role_policy" {
+  statement {
+    sid    = ""
+    effect = "Allow"
+    principals {
+      type = "Service"
+      identifiers = [
+        "ec2.amazonaws.com",
```
Comment (tom-webber) [inline terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf]:
```suggestion
        "eks.amazonaws.com",
```

File: terraform/aws/analytical-platform-data-production/airflow/iam-roles.tf [hunk 2]
```diff
@@ -85,4 +85,15 @@ resource "aws_iam_role" "airflow_prod_execution_role" {
...
+}
+
+########################Airflow dev EKS Role###########################
+
+resource "aws_iam_role" "airflow_dev_eksRole_role_211908c" {
+  name               = "airflow-dev-eksRole-role-211908c"
+  description        = "EKS role for Airflow"
+  assume_role_policy = data.aws_iam_policy_document.airflow_dev_eksRole_role_211908c_assume_role_policy.json
+  managed_policy_arns = [
+    "arn:aws:iam::593291632749:role/airflow-dev-eksRole-role-211908c"
```
Comment (tom-webber) [inline terraform/aws/analytical-platform-data-production/airflow/iam-roles.tf]:
```suggestion
    "arn:aws:iam::aws:policy/AmazonEKSClusterPolicy"
```

**Review by tom-webber — CHANGES_REQUESTED** [review]
changes arising from `terraform plan` output

File: terraform/aws/analytical-platform-data-production/airflow/iam-roles.tf [hunk 3]
```diff
@@ -85,4 +85,15 @@ resource "aws_iam_role" "airflow_prod_execution_role" {
     name   = "airflow-prod-execution-role-policy"
     policy = data.aws_iam_policy_document.airflow_prod_execution_role_policy.json
   }
+}
+
+########################Airflow dev EKS Role###########################
+
+resource "aws_iam_role" "airflow_dev_eksRole_role_211908c" {
+  name               = "airflow-dev-eksRole-role-211908c"
+  description        = "EKS role for Airflow"
```
Comment (tom-webber) [inline terraform/aws/analytical-platform-data-production/airflow/iam-roles.tf]:
```suggestion
  description        = "Allows EKS to manage clusters on your behalf."
```

File: terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf (hunk 1 above)
Comment (tom-webber) [inline terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf]:
```suggestion
        "eks.amazonaws.com",
```

**Review by tom-webber — CHANGES_REQUESTED** [review]

File: terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf (hunk 1 above)
Comment (tom-webber) [inline terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf]:
```suggestion
        "eks.amazonaws.com",
```

File: terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf [hunk 4]
```diff
@@ -334,3 +334,22 @@ data "aws_iam_policy_document" "airflow_prod_execution_role_policy" {
     resources = ["arn:aws:eks:eu-west-1:${var.account_ids["analytical-platform-data-production"]}:cluster/airflow-prod"]
   }
 }
+
+
+########################Airflow dev EKS Policy###########################
+
+data "aws_iam_policy_document" "airflow_dev_eksRole_role_211908c_assume_role_policy" {
```
Comment (tom-webber) [inline terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf]:
```suggestion
data "aws_iam_policy_document" "airflow_dev_eks_role_role_211908c_assume_role_policy" {
```

File: terraform/aws/analytical-platform-data-production/airflow/iam-roles.tf [hunk 5]
```diff
@@ -85,4 +85,15 @@ resource "aws_iam_role" "airflow_prod_execution_role" {
...
     policy = data.aws_iam_policy_document.airflow_prod_execution_role_policy.json
   }
+}
+
+########################Airflow dev EKS Role###########################
+
+resource "aws_iam_role" "airflow_dev_eksRole_role_211908c" {
+  name               = "airflow-dev-eksRole-role-211908c"
+  description        = "Allows EKS to manage clusters on your behalf."
+  assume_role_policy = data.aws_iam_policy_document.airflow_dev_eksRole_role_211908c_assume_role_policy.json
```
Comment (tom-webber) [inline terraform/aws/analytical-platform-data-production/airflow/iam-roles.tf]:
```suggestion
  assume_role_policy = data.aws_iam_policy_document.airflow_dev_eks_role_role_211908c_assume_role_policy.json
```

File: terraform/aws/analytical-platform-data-production/airflow/iam-roles.tf [hunk 6]
```diff
@@ -85,4 +85,15 @@ resource "aws_iam_role" "airflow_prod_execution_role" {
     name   = "airflow-prod-execution-role-policy"
     policy = data.aws_iam_policy_document.airflow_prod_execution_role_policy.json
   }
+}
+
+########################Airflow dev EKS Role###########################
+
+resource "aws_iam_role" "airflow_dev_eksRole_role_211908c" {
+  name               = "airflow-dev-eksRole-role-211908c"
```
Comment (tom-webber) [inline terraform/aws/analytical-platform-data-production/airflow/iam-roles.tf]:
```suggestion
resource "aws_iam_role" "airflow_dev_eks_role" {
  name               = "airflow-dev-eksRole-role-211908c"
```

File: terraform/aws/analytical-platform-data-production/airflow/iam-roles.tf [hunk 7]
```diff
@@ -85,4 +85,15 @@ resource "aws_iam_role" "airflow_prod_execution_role" {
     name   = "airflow-prod-execution-role-policy"
     policy = data.aws_iam_policy_document.airflow_prod_execution_role_policy.json
   }
+}
+
+########################Airflow dev EKS Role###########################
+
+resource "aws_iam_role" "airflow_dev_eks_role_role_211908c" {
```
Comment (tom-webber) [inline terraform/aws/analytical-platform-data-production/airflow/iam-roles.tf]:
```suggestion
resource "aws_iam_role" "airflow_dev_eks_role" {
```

File: terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf [hunk 8]
```diff
@@ -334,3 +334,22 @@ data "aws_iam_policy_document" "airflow_prod_execution_role_policy" {
...
+      type = "Service"
+      identifiers = [
+        "eks.amazonaws.com",
+
+      ]
+    }
+    actions = ["sts:AssumeRole"]
+  }
+
+}
```
Comment (tom-webber) [inline terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf]:
```suggestion
}

```

File: terraform/aws/analytical-platform-data-production/airflow/iam-roles.tf [hunk 9]
```diff
@@ -85,4 +85,15 @@ resource "aws_iam_role" "airflow_prod_execution_role" {
...
+########################Airflow dev EKS Role###########################
+
+resource "aws_iam_role" "airflow_dev_eks_role" {
+  name               = "airflow-dev-eksRole-role-211908c"
+  description        = "Allows EKS to manage clusters on your behalf."
+  assume_role_policy = data.aws_iam_policy_document.airflow_dev_eks_role_role_211908c_assume_role_policy.json
+  managed_policy_arns = [
+    "arn:aws:iam::aws:policy/AmazonEKSClusterPolicy"
+  ]
 }
```
Comment (tom-webber) [inline terraform/aws/analytical-platform-data-production/airflow/iam-roles.tf]:
```suggestion
}

```
fix End Of File

File: terraform/aws/analytical-platform-data-production/airflow/iam-roles.tf [hunk 10]
```diff
@@ -85,4 +85,15 @@ resource "aws_iam_role" "airflow_prod_execution_role" {
...
-}
\ No newline at end of file
+}
+
+########################Airflow dev EKS Role###########################
+
+resource "aws_iam_role" "airflow_dev_eks_role" {
+  name               = "airflow-dev-eksRole-role-211908c"
+  description        = "Allows EKS to manage clusters on your behalf."
+  assume_role_policy = data.aws_iam_policy_document.airflow_dev_eks_role_role_211908c_assume_role_policy.json
```
Comment (tom-webber) [inline terraform/aws/analytical-platform-data-production/airflow/iam-roles.tf]:
```suggestion
  assume_role_policy = data.aws_iam_policy_document.airflow_dev_eks_assume_role_policy.json
```

File: terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf [hunk 11]
```diff
@@ -334,3 +334,22 @@ data "aws_iam_policy_document" "airflow_prod_execution_role_policy" {
     resources = ["arn:aws:eks:eu-west-1:${var.account_ids["analytical-platform-data-production"]}:cluster/airflow-prod"]
   }
 }
+
+
+########################Airflow dev EKS Policy###########################
+
+data "aws_iam_policy_document" "airflow_dev_eks_role_role_211908c_assume_role_policy" {
```
Comment (tom-webber) [inline terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf]:
```suggestion
data "aws_iam_policy_document" "airflow_dev_eks_assume_role_policy" {
```

File: terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf [hunk 12]
```diff
@@ -334,3 +334,22 @@ data "aws_iam_policy_document" "airflow_prod_execution_role_policy" {
...
+      type = "Service"
+      identifiers = [
+        "eks.amazonaws.com",
+
+      ]
+    }
+    actions = ["sts:AssumeRole"]
+  }
+
+}
```
Comment (tom-webber) [inline terraform/aws/analytical-platform-data-production/airflow/iam-policies.tf]:
```suggestion
}

```

**Review by tom-webber — CHANGES_REQUESTED** [review]

**Review by tom-webber — DISMISSED** [review]

**Review by julialawrence — APPROVED** [review]
LGTM

**Pull request merged**